```bash
mypy.....................................................................Failed
- hook id: mypy
- duration: 131.55s
- exit code: 1

pandas/_testing/__init__.py:1051: error: "ExtensionDtype" has no attribute "storage"  [attr-defined]
pandas/_testing/__init__.py:1058: error: "ExtensionDtype" has no attribute "storage"  [attr-defined]
Found 2 errors in 1 file (checked 1445 source files)
```

--- a/pandas/_testing/__init__.py
+++ b/pandas/_testing/__init__.py
@@ -1048,14 +1048,14 @@ def shares_memory(left, right) -> bool:
     if (
         isinstance(left, ExtensionArray)
         and is_string_dtype(left.dtype)
-        and left.dtype.storage in ("pyarrow", "pyarrow_numpy")
+        and left.dtype.storage in ("pyarrow", "pyarrow_numpy")  # type: ignore[attr-defined]  # noqa: E501
     ):
         # https://github.com/pandas-dev/pandas/pull/43930#discussion_r736862669
         left = cast("ArrowExtensionArray", left)
         if (
             isinstance(right, ExtensionArray)
             and is_string_dtype(right.dtype)
-            and right.dtype.storage in ("pyarrow", "pyarrow_numpy")
+            and right.dtype.storage in ("pyarrow", "pyarrow_numpy")  # type: ignore[attr-defined]  # noqa: E501
         ):
             right = cast("ArrowExtensionArray", right)
             left_pa_data = left._pa_array

